mailpoet/assets/js/src/automation/integrations/mailpoet/steps/order_status_changed/edit/order_status.ts

```diff
@@ -0,0 +1,25 @@
+import { OrderStatusOptions } from '../../../../../types/filters';
+import { Hooks } from '../../../../../../hooks';
+
+declare global {
+  interface Window {
+    mailpoet_woo_order_status: Record<string, string>;
+  }
+}
+export const COMPLETED_ORDER_STATUS = 'wc-completed';
+export const orderStatus: OrderStatusOptions = Hooks.applyFilters(
+  'mailpoet.automation.trigger.order_status_changed. ',
+  Object.keys(window.mailpoet_woo_order_status)
+    .map((value: string) => {
+      const status = {
+        value,
+        label: window.mailpoet_woo_order_status[value],
+        isDisabled: value !== COMPLETED_ORDER_STATUS,
+      };
+      return status;
+    })
+    .reduce((acc, curr) => {
+      acc[curr.value] = curr;
+      return acc;
```

Comment:
Looking further, I think a plain JS object has no order of properties defined. That may (or may not 😄) result in strange behavior in some browsers. I think we should either use an array or a `Map` (which preservers order as well).